gpiozero/devices.py

```diff
@@ -380,6 +380,8 @@ def __init__(self, pin=None):
             raise GPIOPinMissing('No pin given')
         if isinstance(pin, int):
             pin = pin_factory(pin)
+        if not isinstance(pin, Pin):
```

Comment:
@waveform80 's more familiar with the Pin internals than me ;-)
With the code on the current master branch, I think this looks fine but it might need to be enhanced when the pin-factory stuff gets merged (which I need to get back to reviewing at some point, when I'm not so busy!)

Only a minor difference, but I'd be tempted to change `if not isinstance(pin, Pin):` to:
```python
elif isinstance(pin, Pin):
  pass
else:
```